packages/babel-traverse/src/path/comments.js

```diff
@@ -5,6 +5,9 @@
  */
 
 export function shareCommentsWithSiblings() {
+  // NOTE: this assumes numbered keys
+  if (typeof this.key === "string") return;
```

Comment:
the key is basically the position in `body` or a `block` via an integer but if it's part of another node it's just the property name of the key (so for for-of it's like left, right, body). For a `LabeledStatement` its an integer too.